src/main/java/io/cdap/plugin/gcp/dataplex/common/util/DataplexUtil.java

```diff
@@ -0,0 +1,414 @@
+    }
+  }
+
+  private static GoogleCredentials getCredentialsFromConfiguration(Configuration configuration) throws IOException {
+    String serviceAccount;
+    String serviceAccountType = configuration.get(DataplexConstants.SERVICE_ACCOUNT_TYPE);
+    Boolean isServiceAccountJson = false;
+    if (serviceAccountType == GCPConnectorConfig.SERVICE_ACCOUNT_JSON) {
+      isServiceAccountJson = true;
+      serviceAccount = configuration.get(SERVICE_ACCOUNT_JSON);
+    } else {
+      serviceAccount = configuration.get(SERVICE_ACCOUNT_KEYFILE);
+    }
+    String filePath = configuration.get(DataplexConstants.SERVICE_ACCOUNT_FILEPATH);
+    return getCredentialsFromServiceAccount(isServiceAccountJson, filePath, serviceAccount);
+  }
+
+  private static GoogleCredentials getCredentialsFromServiceAccount(Boolean isServiceAccountJson, String filePath,
+                                                                    String serviceAccount) throws IOException {
+    if (isServiceAccountJson || (filePath != null && !filePath.equalsIgnoreCase(DataplexConstants.NONE))) {
+      return GCPUtils.loadServiceAccountCredentials(serviceAccount, !isServiceAccountJson)
+        .createScoped(Lists.newArrayList(DataplexConstants.DATAPLEX_SCOPE));
+    } else {
+      return ServiceAccountCredentials.getApplicationDefault().createScoped(
+        Lists.newArrayList(DataplexConstants.DATAPLEX_SCOPE));
+    }
+  }
+
+  /**
+   * Return DataplexServiceClient to call dataplex methods.
+   *
+   * @param credentials GoogleCredentials
+   * @return DataplexServiceClient
+   */
+  public static DataplexServiceClient getDataplexServiceClient(GoogleCredentials credentials) {
+    DataplexServiceSettings dataplexServiceSettings = null;
+    DataplexServiceClient dataplexServiceClient = null;
+    try {
+      dataplexServiceSettings = DataplexServiceSettings.newBuilder()
+        .setCredentialsProvider(FixedCredentialsProvider.create(credentials))
+        .build();
+      dataplexServiceClient = DataplexServiceClient.create(dataplexServiceSettings);
+    } catch (IOException e) {
+      LOG.error("Failed to get DataplexServiceClient. Check the credentials");
+    }
+    return dataplexServiceClient;
+  }
+
+  /**
+   * Return MetadataServiceClient to call dataplex methods.
+   *
+   * @param credentials
+   * @return MetadataServiceClient
+   */
+  public static MetadataServiceClient getMetadataServiceClient(GoogleCredentials credentials) {
+    MetadataServiceSettings metadataServiceSettings = null;
+    MetadataServiceClient metadataServiceClient = null;
+    try {
+      metadataServiceSettings =
+        MetadataServiceSettings.newBuilder()
+          .setCredentialsProvider(FixedCredentialsProvider.create(credentials))
+          .build();
+      metadataServiceClient =
+        MetadataServiceClient.create(metadataServiceSettings);
+    } catch (RuntimeException | IOException e) {
+      LOG.error("Failed to get MetadataServiceClient. Check the credentials");
```

Comment:
should we throw an exception if we're unable to create a MetadataServiceClient? Otherwise we'll return null and caller will get a NullPointerException

Reply:
throwing exception seems to be a better option, will do that.